src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -0,0 +1,184 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog\Product;
+
+use Exception;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\UpdateCombinationFromListingCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Query\GetEditableCombinationsList;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationListForEditing;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class CombinationController extends FrameworkBundleAdminController
+{
+    /**
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param int $productId
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function getListAction(int $productId, Request $request): JsonResponse
+    {
+        $limit = (int) $request->query->get('limit');
+        $page = (int) $request->query->get('page');
+
+        $combinationsList = $this->getQueryBus()->handle(new GetEditableCombinationsList(
+            $productId,
+            $this->getContextLangId(),
+            $limit ?? null,
+            $page ?? null
+        ));
+
+        return $this->json($this->formatResponse($combinationsList));
+    }
+
+    /**
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param int $combinationId
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function updateImpactOnPriceAction(int $combinationId, Request $request): JsonResponse
+    {
+        $impactOnPrice = $request->request->get('impactOnPrice');
+
+        if (!$impactOnPrice) {
+            return $this->json(
+                ['message' => 'Missing impactOnPrice'],
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        $command = new UpdateCombinationFromListingCommand($combinationId);
```

Comment:
Since the `UpdateCombinationFromListingCommand` is now handled separately for each field I wonder if it's really useful Since the commands are all partial we could simply use `UpdateCombinationPricesCommand` here, and `UpdateCombinationStockCommand` in the other ajax request

We would only be missing a new `UpdateProductDefaultCombinationCommand` I guess, but then `UpdateCombinationFromListingCommand` could be removed altogether

Reply:
yeah I guess we can do it in another PR, we've changed the "specs" over and over again so that's what we have now :sweat_smile: